### DISCUSSION TITLE
Decryption of protected pdfs using pdf-lib

### DISCUSSION THREAD

**Discussion by arnoob16** [Hopding/pdf-lib#1432] [Q&A]
I have a password protected pdf file, which can be previewed easily by taking the password as input from the user, but I'm trying to achieve a decryption on the client side, such that, when the user has provided the password for the protected-pdf, I'm able to create a new, decrypted file with the contents of the original encrypted file.

**Comment by sepsol** [comment]
Same here. Were you finally able to find a solution to this?

**Comment by alistair0adams** [comment]
Use this fork https://github.com/cantoo-scribe/pdf-lib

Then do .load(new Uint8Array(blob.getBytes()), { updateMetadata: true, password: "" })

See also https://www.npmjs.com/package/@cantoo/pdf-lib
